trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/credentials/Credentials.java

```diff
@@ -13,22 +13,25 @@
  */
 package io.trino.s3.proxy.server.credentials;
 
+import java.util.Optional;
+
 import static java.util.Objects.requireNonNull;
 
-public record Credentials(Credential emulated, Credential real)
+public record Credentials(Credential emulated, Optional<Credential> real)
 {
     public Credentials
     {
         requireNonNull(emulated, "emulated is null");
         requireNonNull(real, "real is null");
     }
 
-    public record Credential(String accessKey, String secretKey)
+    public Credential requiredRealCredential()
+    {
+        return real.orElseThrow(() -> new IllegalStateException("Credentials do not have real credential"));
```

Comment:
Nitpick: `Credentials pair does not have a configured real credential`? The wording is slightly confusing atm IMO

Reply:
Changed to `Credentials are emulated only and cannot be used for remote access`. OK?